components/screens/MenuScreen.tsx

```diff
@@ -1,30 +1,81 @@
-import {useContext} from 'react';
+import React from 'react';
 
 import {StyleSheet} from 'react-native';
 import {SafeAreaView} from 'react-native-safe-area-context';
 
-import {Heading, HStack, VStack, Switch, Text} from 'native-base';
+import * as Updates from 'expo-updates';
 
-import {ClientContext, ClientProps} from 'clientContext';
-import {AvalancheCenterSelector} from 'components/AvalancheCenterSelector';
+import {Heading, HStack, VStack, Switch, Text, Divider} from 'native-base';
 
-export const MenuScreen = () => {
-  const {staging, toggleStaging, setAvalancheCenter} = useContext<ClientProps>(ClientContext);
+import {AvalancheCenterCard, AvalancheCenterSelector} from 'components/AvalancheCenterSelector';
+
+import {createNativeStackNavigator, NativeStackScreenProps} from '@react-navigation/native-stack';
+
+import {MenuStackParamList, MenuStackNavigationProps} from 'routes';
+import {useNavigation} from '@react-navigation/native';
+
+const MenuStack = createNativeStackNavigator<MenuStackParamList>();
+export const MenuStackScreen = (
+  avalancheCenterId: string,
+  setAvalancheCenter: React.Dispatch<React.SetStateAction<string>>,
+  staging: boolean,
+  setStaging: React.Dispatch<React.SetStateAction<boolean>>,
+) => {
   return (
-    <SafeAreaView style={styles.fullscreen}>
-      <VStack pt="16" px="4" space="4" style={styles.fullscreen}>
-        <Heading>Settings</Heading>
-        <Heading size="sm">Debug settings</Heading>
-        <HStack justifyContent="space-between" alignItems="center" space="4">
-          <Text>Use staging environment</Text>
-          <Switch value={staging} onValueChange={toggleStaging} />
-        </HStack>
-        <AvalancheCenterSelector setAvalancheCenter={setAvalancheCenter} />
-      </VStack>
-    </SafeAreaView>
+    <MenuStack.Navigator initialRouteName="menu">
+      <MenuStack.Screen name="menu" component={MenuScreen(avalancheCenterId, staging, setStaging)} options={{title: `Settings`}} />
+      <MenuStack.Screen
+        name="avalancheCenterSelector"
+        component={AvalancheCenterSelectorScreen(avalancheCenterId, setAvalancheCenter)}
+        options={{title: `Choose An Avalanche Center`}}
```

Comment:
not sure if I am just not understanding the flow, but when I load the app, and go to menu which shows settings (assuming that currently when the app loads, center is already selected for me, which is NWAC, the setting screen shows "choose an avalanche center" and the only option below it is nwac. If I click on nwac, it takes me to the list of all centers. if I select nwac again from this list, it takes me to the main map but it appears to be more zoomed in then before (not sure why)

i assume the experience would be something like: "current avalanche center" and then nwac below, and then below that would be a button that would basically say something like "change center"? but let me know if I am misunderstanding. 

I do like this flow (having the menu/settings determine the center that is), easy to take add new centers as they onboard or take them away too and less advanced users wont be too confused hopefully :D

Reply:
I was modelling this off of the Slack menu where you can choose the app icon - it shows the currently-selected icon in the menu and makes it obvious through a combination of title and chevron that you can go into the sub-menu to choose a different icon. Likely need to look at more menus to figure out a cleaner way to get these concepts across?